### PR TITLE
[[FIX]] Immediately-invoked arrow functions' parameter doesn't need parentheses.

### LINKED ISSUE #2351: [es6] "Bad invocation" when an immediately-invoked arrow function hasn't parenthesis around its argument
This works:
```javascript
/*jshint esnext: true */

((b) => {})(0);
```

This doesn't work: (But is the same of the first example)
```javascript
/*jshint esnext: true */

(b => {})(0);
```

With the second I get this warning: `Bad invocation`

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
Fixes #2351

### CODE REVIEW

File: src/jshint.js [hunk 1]
```diff
@@ -2548,9 +2548,9 @@ var JSHINT = (function() {
           addInternalSrc(left, p[0].value);
         }
       }
-      if (!left.identifier && left.id !== "." && left.id !== "[" &&
-          left.id !== "(" && left.id !== "&&" && left.id !== "||" &&
-          left.id !== "?" && !(state.option.esnext && left["(name)"])) {
+      if (!left.identifier && left.id !== "." && left.id !== "[" && left.value !== "=>" &&
```
Comment (lukeapage) [inline src/jshint.js]: is it safer to use `left.id` instead of `left.value` ? my only concern.

Comment (nicolo-ribaudo) [inline src/jshint.js]: You are right: using `.value`, this code passes the test:
```js
("=>")();
```

**Comment by lukeapage** [comment]
Thanks!

**Comment by lukeapage** [comment]
Committed - https://github.com/jshint/jshint/commit/d261071e7b523549499077bb13e1affb7e25158e

**Pull request closed without merging**